jetty-io/src/main/java/org/eclipse/jetty/io/ssl/SslConnection.java

```diff
@@ -308,10 +309,38 @@ private boolean isHandshakeComplete()
         return state == HandshakeState.SUCCEEDED || state == HandshakeState.FAILED;
     }
 
+    private int getApplicationBufferSize()
+    {
+        SSLSession hsSession = _sslEngine.getHandshakeSession();
+        SSLSession session = _sslEngine.getSession();
+        int size = session.getApplicationBufferSize();
+        if (hsSession == null)
```

Comment:
is it worthwhile doing 
```
if (hsSession == null || hsSession == session)
```
or can they never be the same?

Reply:
Good point, will add that.